ultimatepython/classes/encapsulation.py

```diff
@@ -0,0 +1,117 @@
+"""    
+    Encapsulation is a feature of OOP that allows you to hide the
+    implementation details of a class from its users.
+    Encapsulation allows us to limit the access of certain attributes 
+    within a class. This prevents users from directly accessing and modifying such
+    attributes from outside the class. Instead, users must use methods to access and
+    modify attributes.
+"""
+import random
+
+class BankAccount:
+    def __init__(self, account_holder_name: str):
```

Comment:
Can we skip type hinting for now? If you'd like, we can consider adding type hinting to other modules in future PRs